corehq/apps/analytics/tasks.py

```diff
@@ -0,0 +1,41 @@
+from celery.task import task
+import json
+import requests
+import urllib
+
+from settings import ANALYTICS_IDS
+
+
+def _track_on_hubspot(webuser, properties):
+    """
+    Update or create a new "contact" on hubspot. Record that the user has
+    created an account on HQ.
+
+    properties is a dictionary mapping property names to values.
+    Note that property names must exist on hubspot prior to use.
+    """
+    # TODO: Use OAuth
+    # TODO: Use batch requests / be mindful of rate limit
+    api_key = ANALYTICS_IDS.get('HUBSPOT_API_KEY', None)
+    if api_key and not webuser.is_dimagi:
+        req = requests.post(
+            "https://api.hubapi.com/contacts/v1/contact/createOrUpdate/email/{}".format(urllib.quote(webuser.username)),
+            params={'hapikey': api_key},
+            data=json.dumps(
+                {'properties': [
+                    {'property': k, 'value': v} for k, v in properties.items()
+                ]}
+            ),
+        )
+        req.raise_for_status()
+
+
+@task(queue='background_queue')
```

Comment:
Is this what you had in mind? http://celery.readthedocs.org/en/latest/userguide/tasks.html#Task.acks_late
The API calls are idempotent.